contracts/utils/cryptography/ECDSA.sol

```diff
@@ -98,9 +98,12 @@ library ECDSA {
      * _Available since v4.3._
      */
     function tryRecover(bytes32 hash, bytes32 r, bytes32 vs) internal pure returns (address, RecoverError) {
-        bytes32 s = vs & bytes32(0x7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff);
-        uint8 v = uint8((uint256(vs) >> 255) + 27);
-        return tryRecover(hash, v, r, s);
+        unchecked {
+            bytes32 s = vs & bytes32(0x7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff);
```

Comment:
Does it change anything to have this line inside the unchecked block ?

Reply:
No but I thought it would be cleaner like that.

Reply:
We can't have it only around `uint8 v = uint8((uint256(vs) >> 255) + 27);`, because that would limit the visibility of `v` and we would not be able to return it.

Reply:
I mean, as an alternative, if you're open to it we can use inline assembly. Something like:

```solidity
uint8 v;
/// @solidity memory-safe-assembly
assembly {
    v := add(shr(255, vs), 27)
}
```